File: frontend/projects/api/src/lib/lifecycle-history.ts
import type { CurriculumStatus } from './curriculum-status';
import type { LifecycleAuditEntry } from './lifecycle-audit-entry';

export interface LifecycleHistory {
  importId: string;
  packageId: string;
  packageVersion: string;
  packageChecksum: string;
  currentStatus: CurriculumStatus;
  correlationId: string;
  entries: LifecycleAuditEntry[];
}

File: backend/src/RepoFluent.Application/LifecycleHistory.cs
using RepoFluent.Domain;

namespace RepoFluent.Application;

public sealed record LifecycleHistory(
    Guid ImportId,
    string PackageId,
    string PackageVersion,
    string PackageChecksum,
    CurriculumStatus CurrentStatus,
    string CorrelationId,
    IReadOnlyList<LifecycleAuditEntry> Entries);
